### PR TITLE
[Experimental] Add HTK SurfaceMeshesToPlane Script

### PR DESCRIPTION
**Pull request by srinjoym**
## Overview
This PR adds a new version of the `SurfaceMeshesToPlane` from HTK as a Utility. This is primaritly useful for users on Hololens 1 as Scene Understanding is not supported on that platform. In a follow up PR, I'll add a `SurfacePlaneObserver` to use this script with the Spatial Awareness system.

This PR also requires the Planefinding DLL to work. We are still setting up CI on the http://github.com/microsoft/mixedrealitytoolkit repo to generate Nuget packages so we can automatically download the DLL with MRTK here.

## Changes
- Fixes: #6246

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        [Range(0.0f, 1.0f)]
+        public float CeilingBuffer = 0.1f;
+
+        [Tooltip("Thickness of rendered plane objects")]
+        [Range(0.0f, 1.0f)]
+        public float PlaneThickness = 0.01f;
+
+        /// <summary>
+        /// Determines which plane types should be rendered.
+        /// </summarydrawPlanesMask
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/PlaneFinding.cs [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+
+            return PinObject(reusedImportedMeshesForMarshalling);
+        }
+
+        /// <summary>
+        /// Marshals BoundedPlane data returned from a DLL API call into a managed BoundedPlane array
+        /// and then frees the memory that was allocated within the DLL.
+        /// </summary>
+        /// <remarks>Disabling warning 618 when calling Marshal.SizeOf(), because
+        /// Unity does not support .Net 4.5.1+ for using the preferred Marshal.SizeOf(T) method."/>, </remarks>
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/PlaneFinding.cs]:
```suggestion
        /// Unity does not support .Net 4.5.1+ for using the preferred Marshal.SizeOf(T) method."/></remarks>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+        public SpatialAwarenessSurfaceTypes Type;
+    }
+
+    /// <summary>
+    /// SurfaceMeshesToPlanes will find and create planes based on the meshes returned by the SpatialMappingManager's Observer.
+    /// </summary>
+    public class SurfaceMeshesToPlanes:MonoBehaviour
+    {
+        [Tooltip("Empty game object used to contain all planes created by the SurfaceToPlanes class")]
+        public GameObject PlanesParent;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]: Please convert these to [private fields with public properties (if needed)]( https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#encapsulation).

This wasn't a guideline we always followed in HTK, if that's where these are coming from, but we're enforcing for MRTK v2.

Comment (srinjoym) [reply Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]: Thanks for pointing that out @keveleigh! Just updated all the properties to be private and have public accessors

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 4]
```diff
@@ -0,0 +1,453 @@
...
+            set => physicsLayer = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Material used to render planes")]
+        private Material defaultMaterial;
+
+        /// <summary>
+        /// Material used to render planes
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 5]
```diff
@@ -0,0 +1,453 @@
...
+            set => defaultMaterial = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Minimum area required for a plane to be created.")]
+        private float minArea = 0.025f;
+
+        /// <summary>
+        /// Minimum area required for a plane to be created.
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+        }
+
+        [SerializeField]
+        [Tooltip("Threshold for acceptable normals (the closer to 1, the stricter the standard). Used when determining plane type.")]
+        [Range(0.0f, 1.0f)]
+        private float upNormalThreshold = 0.9f;
+
+        /// <summary>
+        /// Threshold for acceptable normals (the closer to 1, the stricter the standard). Used when determining plane type.
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+        }
+
+        [SerializeField]
+        [Tooltip("Buffer to use when determining if a horizontal plane near the floor should be considered part of the floor.")]
+        [Range(0.0f, 1.0f)]
+        private float floorBuffer = 0.1f;
+
+        /// <summary>
+        /// Buffer to use when determining if a horizontal plane near the floor should be considered part of the floor.
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+        }
+
+        [SerializeField]
+        [Tooltip("Buffer to use when determining if a horizontal plane near the ceiling should be considered part of the ceiling.")]
+        [Range(0.0f, 1.0f)]
+        private float ceilingBuffer = 0.1f;
+
+        /// <summary>
+        /// Buffer to use when determining if a horizontal plane near the ceiling should be considered part of the ceiling.
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 9]
```diff
@@ -0,0 +1,453 @@
...
+        }
+
+        [SerializeField]
+        [Tooltip("Thickness of rendered plane objects")]
+        [Range(0.0f, 1.0f)]
+        private float planeThickness = 0.01f;
+
+        /// <summary>
+        /// Thickness of rendered plane objects
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

File: Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs [hunk 10]
```diff
@@ -0,0 +1,453 @@
...
+            set => planeThickness = value;
+        }
+
+        [HideInInspector]
+        private SpatialAwarenessSurfaceTypes drawPlanesMask =
+            (SpatialAwarenessSurfaceTypes.Wall | SpatialAwarenessSurfaceTypes.Floor | SpatialAwarenessSurfaceTypes.Ceiling | SpatialAwarenessSurfaceTypes.Platform);
+
+        /// <summary>
+        /// Determines which plane types should be rendered
+        /// </summary
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/Experimental/PlaneFinding/SurfaceMeshesToPlanes.cs]:
```suggestion
        /// </summary>
```

**Comment by davidkline-ms** [comment]
Adapted for inclusion in #9225. Closing this one.

**Pull request closed without merging**